vars/unitTest.groovy

```diff
@@ -84,6 +87,10 @@ def call(Map config = [:]) {
     stashes.add("${target_compiler}-install")
     stashes.add("${target_compiler}-build-vars")
   }
+  
+  if (stage_info['valgrind']) {
+    test_script = test_script.trim() + " " + stage_info['valgrind']
```

Comment:
You cannot/should not do this.  `test_script` holds a path to the test script.  It should not be depended on that these scripts take arguments.  Environment variables is the way we pass in values to these scripts.

Reply:
okay, I'm working on this

Reply:
Hi @brianjmurrell , I've been experimenting with this and I realized that environment variables are shared among stages. For example, in here https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-3023/48/pipeline/442 the complete log of `run_test.sh with memcheck` and `run_test.sh` have the same value for `WITH_VALGRIND`, I think it's a race condition of the last stage that set the variable, and then both stages end up executing the same branch of the `run_test.sh` script

I'm going to try using this, though I don't know if that will work https://wiki.jenkins.io/pages/viewpage.action?pageId=70878639 let me know if you have any suggestions on how to solve this problem

Reply:
> I realized that environment variables are shared among stages

Well, to some degree, yes.  Depending on how you are setting/using them.  Is the problem the evaluation of env.WITH_VALGRIND here:https://github.com/daos-stack/pipeline-lib/blob/1ffb977116ef4fed07f3ce746e42c78d409b9f29/vars/unitTest.groovy#L95-L105?  If so, why do you want to use an environment variable there?  Why not just use the `config` `Map` value?

Reply:
right, I didn't need to set up that variable here. I moved it to runTest and it is not running the same stage twice anymore. I quite don't understand why the problem is solved, but the WITH_VALGRIND variable is not shared in that part of the groovy code